curl-static-musl/plan.sh

```diff
@@ -0,0 +1,53 @@
+
+source ../curl/plan.sh
+
+pkg_name=curl-static-musl
+pkg_distname=curl
```

Comment:
Ok. I had copied one of the other `-static` plans so didn't include it. Looks like I would need to include a bunch of others as well 

https://travis-ci.org/habitat-sh/core-plans/builds/205054250#L488-L500